### ISSUE TITLE
Multiple replacements on a single string will throw error when the matched is the end of the content

### ISSUE THREAD

**Issue by djyde** [iansinnott/react-string-replace#4]
```javascript
let replacedContent
const test = '@username http://a_photo.jpg'
replacedContent = reactStringReplace(test, /(http?:\/\/.*\.(?:png|jpg))/g, (match, i) => (
  <Text key={i}>Image</Text>
))

replacedContent = reactStringReplace(replacedContent, /@(\w+)/g, (match, i) => (
  <Text key={i} style={style.usernameRef}>@{match}</Text>
))
```

Will throw `First argument to react-string-replace#replaceString must be a string`

**Comment by iansinnott** [comment]
Thanks for the repro code @djyde. I was able to repro on my system so I'll be looking into this.

**Comment by iansinnott** [comment]
Should be fixed in 0.3.1

**Comment by djyde** [comment]
Nice work!

**Comment by djyde** [comment]
`const test = 'http://a_photo.jpg some string'` will throw error too

**Comment by iansinnott** [comment]
Damn. @djyde can you give a full repro example as you did with the initial comment?

**Comment by iansinnott** [comment]
I tried the following and it did not throw. If you can give me an example snippet that will throw that would hep me debug.

```js
const string = 'http://a_photo.jpg some string';
const replacedContent = replaceString(string, /(http?:\/\/.*\.(?:png|jpg))/g, match => {
  return { key: 'image', match };
});
```

**Comment by djyde** [comment]
Occur when multiple replacements

```
let replacedContent
const test = 'http://a_photo.jpg some string'
replacedContent = reactStringReplace(test, /(http?:\/\/.*\.(?:png|jpg))/g, (match, i) => (
  <Text key={i}>Image</Text>
))

replacedContent = reactStringReplace(replacedContent, /@(\w+)/g, (match, i) => (
  <Text key={i} style={style.usernameRef}>@{match}</Text>
))
```

**Comment by iansinnott** [comment]
It seems I was getting a bit overzealous with my handling of what I thought would be invalid inputs. I removed the input check that was throwing that error. Hopefully it works now. The fix is in 0.3.2.
